pkg/commonvpc/vpc.go

```diff
@@ -179,15 +189,68 @@ func (s *Subnet) AddressRange() *ipblock.IPBlock {
 	return s.IPblock
 }
 func (s *Subnet) IsPrivate() bool {
-	return s.isPrivate
+	return s.subnetExpose == privateExpose
 }
 func (s *Subnet) SetIsPrivate(isPrivate bool) {
-	s.isPrivate = isPrivate
+	s.subnetExpose = publicExpose
+	if isPrivate {
+		s.subnetExpose = privateExpose
+	}
 }
 func (s *Subnet) SynthesisKind() spec.ResourceType {
 	return spec.ResourceTypeSubnet
 }
 
+// ////////////////////////////////////////////////////////////////////////////////
+// privateSubnetRule is the implementation of PrivateSubnetRule
+// it holds the information on the influence of the subnet on the connectivity.
+// its relevant only for providers that allow private subnets (aws)
+type privateSubnetRule struct {
+	subnet    vpcmodel.Subnet
+	src, dst  vpcmodel.Node
+	isIngress bool
+}
+
+func newPrivateSubnetRule(subnet vpcmodel.Subnet, src, dst vpcmodel.Node, isIngress bool) vpcmodel.PrivateSubnetRule {
+	return &privateSubnetRule{subnet, src, dst, isIngress}
+}
+
+func (psr *privateSubnetRule) Deny(isIngress bool) bool {
+	return isIngress == psr.isIngress && psr.subnet.IsPrivate()
+}
+
+func (psr *privateSubnetRule) String() string {
```

Comment:
I'll finalize the wording and format here in my PR which will link the rule to explainabity

Reply:
tnx